zio-sbt-ci/src/main/scala/zio/sbt/ZioSbtCiPlugin.scala

```diff
@@ -177,7 +178,7 @@ object ZioSbtCiPlugin extends AutoPlugin {
                   scalaVersionMatrix.values.toSeq.flatten.distinct.map { scalaVersion: String =>
                     Step.SingleStep(
                       name = "Test",
-                      condition = Some(Condition.Expression(s"matrix.scala == '$scalaVersion'")),
+                      `if` = Some(Condition.Expression(s"matrix.scala == '$scalaVersion'")),
```

Comment:
Can we preserve the previous syntax? We can deprecate it, but make it possible to compile old codes.

Reply:
yes, I made a custom codec for `Seq[Step]` in `Job`

Reply:
I tried but because of all the default values there are already issues with overloaded constructors. If the zio-json PR will get merged I could transform a lot of optional sequences to plain sequences with the `explicitEmptyCollections` set to false.